freedomofpress/securedrop
# Nested Virtualization of Staging Env in CI

## Status

Ready for review

## Description of Changes

Fixes #3702 

Changes proposed in this pull request:

* Drops AWS staging logic
* Introduces GCE staging logic
* Now with :100: more GRSEC tests (previously there were none)

## Testing

How should the reviewer test this PR?

To replicate CI env locally:

* Get access to Google service account credentials (talk to @msheiny :heart_eyes: )
* export that service account as an environment variable called `GOOGLE_CREDENTIALS`
* source ci env file (`. devops/gce-nested/ci-env.sh`)
* Start the environment (`devops/gce-nested/gce-start.sh`)
* Start the test (`devops/gce-nested/gce-runner.sh`)
* Kill the test (`devops/gce-nested/gce-stop.sh`)

1 GRSEC tests is failing and has been `xfail`ed. Making a new ticket to track fix for that. The problem is reproducible locally and existed before introduction of this PR.

## Deployment

Any special considerations for deployment? Consider both:

Only affects CI
